src/awkward/_v2/cpp-headers/awkward/utils.h

```diff
@@ -87,6 +89,18 @@ type_to_name<char>() {
     return "char";
 }
 
+template <>
+const std::string
+type_to_name<std::complex<float>>() {
+    return "complex64";
+}
+
+template <>
+const std::string
+type_to_name<std::complex<double>>() {
+    return "complex128";
+}
```

Comment:
Explicitly setting these is a good idea. If the types other than complex are assigned through `typeid(T).name()`, it would be better to make them explicit overrides, too. The typeid name is platform-dependent: it might come out to be the right name on one system, but it then be the wrong name on another.

Reply:
@jpivarski All the other primitive types are set explicitly too. The `typeid(T).name()` is just for the case user tries to use a type other than the primitives.

Reply:
Okay, good! I forgot that I was looking at a diff.